library/src/main/java/org/apache/apex/malhar/lib/window/impl/AbstractWindowedJoinOperator.java

```diff
@@ -0,0 +1,206 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.apex.malhar.lib.window.impl;
+
+import java.util.Collection;
+import java.util.Collections;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.apache.apex.malhar.lib.window.ControlTuple;
+import org.apache.apex.malhar.lib.window.JoinAccumulation;
+import org.apache.apex.malhar.lib.window.JoinWindowedOperator;
+import org.apache.apex.malhar.lib.window.Tuple;
+import org.apache.apex.malhar.lib.window.Window;
+import org.apache.apex.malhar.lib.window.WindowOption;
+import org.apache.apex.malhar.lib.window.WindowState;
+import org.apache.apex.malhar.lib.window.WindowedStorage;
+
+import com.google.common.base.Function;
+import com.datatorrent.api.DefaultInputPort;
+import com.datatorrent.api.annotation.InputPortFieldAnnotation;
+
+
+/**
+ * Abstract Windowed Join Operator.
+ */
+public abstract class AbstractWindowedJoinOperator<InputT1, InputT2, OutputT, DataStorageT extends WindowedStorage,
+    RetractionStorageT extends WindowedStorage, AccumulationT extends
+    JoinAccumulation>
+    extends AbstractWindowedOperator<InputT1, OutputT, DataStorageT, RetractionStorageT, AccumulationT>
+    implements JoinWindowedOperator<InputT1, InputT2>
+{
+
+  private static final transient Logger LOG = LoggerFactory.getLogger(AbstractWindowedJoinOperator.class);
+  private Function<InputT2, Long> timestampExtractor2;
+
+  private long latestWatermark1 = -1;  // latest watermark from stream 1
+  private long latestWatermark2 = -1;  // latest watermark from stream 2
+
+
+  private static final transient Collection<? extends Window> GLOBAL_WINDOW_SINGLETON_SET = Collections.singleton(Window.GlobalWindow.getInstance());
```

Comment:
I believe this is already declared somewhere else. If so, we should use that instead.